local of_typeof = require("./utility/of_typeof")

return {
	of_instancetype = require("@self/of_instancetype"),
	of_instanceisa = require("@self/of_instanceisa"),
	of_enumtype = require("@self/of_enumtype"),
	as_warn = require("@self/as_warn"),

	Axes = of_typeof("Axes"),
	BrickColor = of_typeof("BrickColor"),
	CatalogSearchParams = of_typeof("CatalogSearchParams"),
	CFrame = of_typeof("CFrame"),
	Color3 = of_typeof("Color3"),
	ColorSequence = of_typeof("ColorSequence"),
	ColorSequenceKeypoint = of_typeof("ColorSequenceKeypoint"),
	Content = of_typeof("Content"),
	DateTime = of_typeof("DateTime"),
	DockWidgetPluginGuiInfo = of_typeof("DockWidgetPluginGuiInfo"),
	Enum = of_typeof("Enum"),
	EnumItem = of_typeof("EnumItem"),
	Enums = of_typeof("Enums"),
	Faces = of_typeof("Faces"),
	FloatCurveKey = of_typeof("FloatCurveKey"),
	Font = of_typeof("Font"),
	Instance = of_typeof("Instance"),
	NumberRange = of_typeof("NumberRange"),
	NumberSequence = of_typeof("NumberSequence"),
	NumberSequenceKeypoint = of_typeof("NumberSequenceKeypoint"),
	OverlapParams = of_typeof("OverlapParams"),
	Path2DControlPoint = of_typeof("Path2DControlPoint"),
	PathWaypoint = of_typeof("PathWaypoint"),
	PhysicalProperties = of_typeof("PhysicalProperties"),
	Random = of_typeof("Random"),
	Ray = of_typeof("Ray"),
	RaycastParams = of_typeof("RaycastParams"),
	RaycastResult = of_typeof("RaycastResult"),
	RBXScriptConnection = of_typeof("RBXScriptConnection"),
	RBXScriptSignal = of_typeof("RBXScriptSignal"),
	Rect = of_typeof("Rect"),
	Region3 = of_typeof("Region3"),
	Region3int16 = of_typeof("Region3int16"),
	RotationCurveKey = of_typeof("RotationCurveKey"),
	Secret = of_typeof("Secret"),
	SecurityCapabilities = of_typeof("SecurityCapabilities"),
	SharedTable = of_typeof("SharedTable"),
	TweenInfo = of_typeof("TweenInfo"),
	UDim = of_typeof("UDim"),
	UDim2 = of_typeof("UDim2"),
	ValueCurveKey = of_typeof("ValueCurveKey"),
	Vector2 = of_typeof("Vector2"),
	Vector2int16 = of_typeof("Vector2int16"),
	Vector3 = of_typeof("Vector3"),
	Vector3int16 = of_typeof("Vector3int16"),
}
