local of_typeof = require("./utility/of_typeof")

return {
	of_instancetype = require("@self/of_instancetype"),
	of_instanceisa = require("@self/of_instanceisa"),
	of_enumtype = require("@self/of_enumtype"),
	as_warn = require("@self/as_warn"),

	Axes = of_typeof<<Axes>>("Axes"),
	BrickColor = of_typeof<<BrickColor>>("BrickColor"),
	CatalogSearchParams = of_typeof<<CatalogSearchParams>>("CatalogSearchParams"),
	CFrame = of_typeof<<CFrame>>("CFrame"),
	Color3 = of_typeof<<Color3>>("Color3"),
	ColorSequence = of_typeof<<ColorSequence>>("ColorSequence"),
	ColorSequenceKeypoint = of_typeof<<ColorSequenceKeypoint>>("ColorSequenceKeypoint"),
	Content = of_typeof<<Content>>("Content"),
	DateTime = of_typeof<<DateTime>>("DateTime"),
	DockWidgetPluginGuiInfo = of_typeof<<DockWidgetPluginGuiInfo>>("DockWidgetPluginGuiInfo"),
	Enum = of_typeof<<Enum>>("Enum"),
	EnumItem = of_typeof<<EnumItem>>("EnumItem"),
	Enums = of_typeof<<Enums>>("Enums"),
	Faces = of_typeof<<Faces>>("Faces"),
	FloatCurveKey = of_typeof<<FloatCurveKey>>("FloatCurveKey"),
	Font = of_typeof<<Font>>("Font"),
	Instance = of_typeof<<Instance>>("Instance"),
	NumberRange = of_typeof<<NumberRange>>("NumberRange"),
	NumberSequence = of_typeof<<NumberSequence>>("NumberSequence"),
	NumberSequenceKeypoint = of_typeof<<NumberSequenceKeypoint>>("NumberSequenceKeypoint"),
	OverlapParams = of_typeof<<OverlapParams>>("OverlapParams"),
	Path2DControlPoint = of_typeof<<Path2DControlPoint>>("Path2DControlPoint"),
	PathWaypoint = of_typeof<<PathWaypoint>>("PathWaypoint"),
	PhysicalProperties = of_typeof<<PhysicalProperties>>("PhysicalProperties"),
	Random = of_typeof<<Random>>("Random"),
	Ray = of_typeof<<Ray>>("Ray"),
	RaycastParams = of_typeof<<RaycastParams>>("RaycastParams"),
	RaycastResult = of_typeof<<RaycastResult>>("RaycastResult"),
	RBXScriptConnection = of_typeof<<RBXScriptConnection>>("RBXScriptConnection"),
	RBXScriptSignal = of_typeof<<RBXScriptSignal>>("RBXScriptSignal"),
	Rect = of_typeof<<Rect>>("Rect"),
	Region3 = of_typeof<<Region3>>("Region3"),
	Region3int16 = of_typeof<<Region3int16>>("Region3int16"),
	RotationCurveKey = of_typeof<<RotationCurveKey>>("RotationCurveKey"),
	Secret = of_typeof<<Secret>>("Secret"),
	SecurityCapabilities = of_typeof<<SecurityCapabilities>>("SecurityCapabilities"),
	SharedTable = of_typeof<<SharedTable>>("SharedTable"),
	TweenInfo = of_typeof<<TweenInfo>>("TweenInfo"),
	UDim = of_typeof<<UDim>>("UDim"),
	UDim2 = of_typeof<<UDim2>>("UDim2"),
	ValueCurveKey = of_typeof<<ValueCurveKey>>("ValueCurveKey"),
	Vector2 = of_typeof<<Vector2>>("Vector2"),
	Vector2int16 = of_typeof<<Vector2int16>>("Vector2int16"),
	Vector3 = of_typeof<<Vector3>>("Vector3"),
	Vector3int16 = of_typeof<<Vector3int16>>("Vector3int16"),
}
